reframe/core/systems.py

```diff
@@ -13,6 +13,169 @@
 from reframe.core.environments import (Environment, ProgEnvironment)
 
 
+class Processor(jsonext.JSONSerializable):
+    '''A representation of a processor inside ReFrame.
+
+    .. warning::
+       Users may not create :class:`Processor` objects directly.
+    '''
+
+    def __init__(self, processor_info):
+        self._arch = None
+        self._num_cpus = None
+        self._num_cpus_per_core = None
+        self._num_cpus_per_socket = None
+        self._num_sockets = None
+        self._topology = None
+        self._info = processor_info
+
+        if processor_info == {}:
+            return
+
+        for var in ['arch', 'num_cpus', 'num_cpus_per_core',
+                    'num_cpus_per_socket', 'num_sockets', 'topology']:
+            if var in processor_info:
+                setattr(self, f'_{var}', processor_info[var])
+
+    @property
+    def info(self):
+        '''All the available information from the configuration.
+
+        :type: :class:`dict`
+        '''
+        return self._info
+
+    @property
+    def arch(self):
+        '''The microarchitecture of the processors.
+
+        :type: :class:`str`
+        '''
+        return self._arch
+
+    @property
+    def num_cpus(self):
+        '''Number of logical CPUs.
+
+        :type: integral
+        '''
+        return self._num_cpus
+
+    @property
+    def num_cpus_per_core(self):
+        '''Number of logical CPUs per core.
+
+        :type: integral
+        '''
+        return self._num_cpus_per_core
+
+    @property
+    def num_cpus_per_socket(self):
+        '''Number of logical CPUs per socket.
+
+        :type: integral
+        '''
+        return self._num_cpus_per_socket
+
+    @property
+    def num_sockets(self):
+        '''Number of sockets.
+
+        :type: integral
+        '''
+        return self._num_sockets
+
+    @property
+    def topology(self):
+        '''Topology of the architecture.
```

Comment:
```suggestion
        '''Processor topology.
```

And we should expand this with an example.